lib/datadog/core/remote/configuration/content.rb

```diff
@@ -17,11 +18,30 @@ def parse(hash)
             end
           end
 
-          attr_reader :path, :data
+          attr_reader :path, :data, :hashes
 
           def initialize(path:, data:)
             @path = path
             @data = data
+            @hashes = {}
+          end
+
+          def hash(type)
+            @hashes[type] || compute_and_store_hash(type)
+          end
+
+          def length
+            @data.size
+          end
+
+          private
+
+          def compute_and_store_hash(type)
+            result = Digest.hexdigest(type, @data)
+            @hashes[type] = result
+            result
+          ensure
+            @data.rewind
           end
```

Comment:
I doubted between having in on `Digest.hexdigest` or on `compute_and_store_hash` In the end I decided to have on `compute_and_store_hash` because it is more obvious to the person reading what `compute_and_store_hash` does, and it makes sure that data is rewind. 
On the other hand, I thought that `Digest.hexdigest` is the one doing the `data.read` calls so should the one responsible to call `data.rewind`

I see that having `data.rewind` on `compute_and_store_hash ` makes `Digest.hexdigest` an incomplete method, since if we call it outside of `compute_and_store_hash` we would have to call `data.rewind`

All of this to say, that yes let's move it to `Digest.hexdigest`